service/src/main/java/org/cbioportal/genome_nexus/service/internal/CancerHotspotServiceImpl.java

```diff
@@ -36,37 +36,40 @@
 import org.cbioportal.genome_nexus.model.*;
 import org.cbioportal.genome_nexus.persistence.HotspotRepository;
 import org.cbioportal.genome_nexus.service.CancerHotspotService;
+import org.cbioportal.genome_nexus.service.GenomicLocationAnnotationService;
 import org.cbioportal.genome_nexus.service.VariantAnnotationService;
-import org.cbioportal.genome_nexus.component.annotation.NotationConverter;
 import org.cbioportal.genome_nexus.service.exception.CancerHotspotsWebServiceException;
 import org.cbioportal.genome_nexus.service.exception.VariantAnnotationNotFoundException;
 import org.cbioportal.genome_nexus.service.exception.VariantAnnotationWebServiceException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import java.util.*;
+import java.util.function.Function;
 
 /**
  * @author Selcuk Onur Sumer
  */
 @Service
 public class CancerHotspotServiceImpl implements CancerHotspotService
 {
+
     private final HotspotRepository hotspotRepository;
     private final VariantAnnotationService variantAnnotationService;
     private final HotspotFilter hotspotFilter;
-    private final NotationConverter notationConverter;
+
+    private final GenomicLocationAnnotationService genomicLocationAnnotationService;
 
     @Autowired
     public CancerHotspotServiceImpl(HotspotRepository hotspotRepository,
                                     VariantAnnotationService verifiedHgvsVariantAnnotationService,
-                                    HotspotFilter hotspotFilter,
-                                    NotationConverter notationConverter)
+                                    GenomicLocationAnnotationService verifiedGenomicLocationAnnotationServiceImpl,
```

Comment:
To keep the consistency, we can rename this to `verifiedGenomicLocationAnnotationService`

Reply:
@leexgh it's not gonna work. There are 2 implementations of `GenomicLocationAnnotationService` interface: `GenomicLocationAnnotationServiceImpl` and `VerifiedGenomicLocationAnnotationServiceImpl`. Spring's `@Autowired` falls back on selecting an implementation based on exact naming when there are multiple implementation of the type.